Content.Server/Power/EntitySystems/MaintenanceOilSystem.cs

```diff
@@ -0,0 +1,118 @@
+using Content.Server.Chemistry.Components.SolutionManager;
+using Content.Server.Chemistry.EntitySystems;
+using Content.Server.NodeContainer.Nodes;
+using Content.Server.NodeContainer;
+using Content.Server.Power.Components;
+using Content.Server.Power.NodeGroups;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.FixedPoint;
+using Content.Shared.Power;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.Random;
+
+namespace Content.Server.Power.EntitySystems;
+
+[UsedImplicitly]
+sealed class MaintenanceOilSystem : EntitySystem
+{
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+    private float _updateTimer = 0.0f;
+    private const float UpdateTime = 10.0f; // 10 seconds. Can't be much lower or FixedPoint2 isn't precise enough.
```

Comment:
Don't store stateful data on systems.

Reply:
This state is used for the system to keep track of when to periodically update. The same pattern is used by ToolSystem.Welder, BarotraumaSystem, FlammableSystem, AdminLogManager, CrewMonitoringConsoleSystem, PowerMonitoringConsoleSystem, SmokingSystem, AdvertiseSystem, and ResearchSystem. Can you suggest an alternative?

Reply:
store it on the component, in the update loop you can subtract / add time and compare on a component basis. This would also allow update staggering in the future if desired. take a look at BloodstreamSystem or GasAnalyzerSystem (I based it on BloodstreamSystem) as those both do this

Reply:
It's a trivial change, and if you insist I'll go ahead and make it. Staggering is not necessary for this system since it is probabilistic on a substation-to-substation basis, anyway. Looping through all substations and and individually checking and bumping their update time seems unnecessary and wasteful, especially since this only needs to update once every 10 seconds.